Cargo.toml

```diff
@@ -0,0 +1,71 @@
+[workspace]
+resolver = "2"
+members = [
+  "src/lib/crypto/kimchi_bindings/stubs",
+  "src/lib/crypto/kimchi_bindings/wasm"
+]
+
+[workspace.dependencies]
+array-init = "2.0.0"
+base64 = "0.13.0"
+console_error_panic_hook = { version = "0.1.6" }
+getrandom = { version = "0.2", features = ["js"] }
+js-sys = "0.3"
+libc = "0.2.62"
+num-bigint = { version = "0.4.4", features = [ "rand", "serde" ] }
+# ocaml-specific
+ocaml = { version = "0.22.2", features = ["no-caml-startup"] }
+ocaml-gen = { version = "0.1.5" }
+once_cell = "1.10.0"
+paste = "1.0.5"
+# Version for proc-macro2 needs to be 1.0.60+ to be compatible with newer rust
+# versions https://github.com/rust-lang/rust/issues/113152
+proc-macro2 = { version = "=1.0.66", features = ["default", "proc-macro"] }
+quote = "1.0.31"
+rand = "0.8.5"
+rayon = "1.5.0"
+rmp-serde = "1.1.2"
+serde = "1.0.171"
+serde_json = "1.0.103"
+serde-wasm-bindgen = ">=0.4"
+serde_with = ">=2.1.0"
+sprs = { version = "0.11.0", features = ["multi_thread"] }
+spmc = { version = "0.3.0" }
+# Strictly enforcing 0.2.87
+# Check it is up-to-date with the version in proof-systems/Cargo.toml
+wasm-bindgen = { version = "=0.2.87" }
+wasm-bindgen-test = ">=0.3.0"
+web-sys = { version = "0.3.35", features = [
+  "Window",
+  "Document",
+  "HtmlElement",
+  "Text",
+  "Node",
+  "Element",
+] }
+
+# arkworks
+ark-ec = { version = "0.4.2", features = ["parallel"] }
+ark-ff = { version = "0.4.2", features = ["parallel", "asm"] }
+ark-poly = { version = "0.4.2", features = ["parallel"] }
+ark-serialize = "0.4.2"
+
+# proof-systems
+groupmap = { git = "https://github.com/o1-labs/proof-systems", rev = "0c64141221529455d1dfcd72570c9f40834c1280" }
```

Comment:
Why is there a concrete fixed revision here? To make sure the same sublibrary is used across all of the related imports?

Reply:
It was replacing the submodule. However, I had to reintroduce it, see de71aa218db6d87fba49f827ccb0123907f4b7c5 and 79179eaf22a095b06dc4ff0ce10103a00cdaadd2.